src/Tizen.NUI/src/public/Theme/ThemeManager.cs

```diff
@@ -0,0 +1,330 @@
+        /// </summary>
+        /// <param name="fallbackTheme">The theme instance to be applied as a fallback.</param>
+        /// <exception cref="ArgumentNullException">The given theme is null.</exception>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public static void ApplyFallbackTheme(Theme fallbackTheme)
+        {
+            DefaultTheme = fallbackTheme ?? throw new ArgumentNullException("Invalid theme.");
+        }
+
+        /// <summary>
+        /// Apply theme to the NUI.
+        /// This will change the appreance of the existing components with property <seealso cref="View.ThemeChangeSensitive"/> on.
+        /// This also affects all components created afterwards.
+        /// </summary>
+        /// <param name="theme">The theme instance to be applied.</param>
+        /// <exception cref="ArgumentNullException">Thrown when the given theme is null.</exception>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public static void ApplyTheme(Theme theme)
+        {
+            var newTheme = (Theme)theme?.Clone() ?? throw new ArgumentNullException("Invalid theme.");
+
+            if (string.IsNullOrEmpty(newTheme.Id))
+            {
+                newTheme.Id = "NONAME";
+            }
+
+            CurrentTheme = newTheme;
+        }
+
+        /// <summary>
+        /// Note that this API is to support legacy Tizen.NUI.Components.StyleManager.
+        /// Please use <seealso cref="ApplyTheme(Theme)"/> instead.
+        ///
+        /// Apply theme to the NUI using theme id.
+        /// The id of theme should be either a registered custom theme or a built-in theme.
+        /// You can register custom theme using <seealso cref="RegisterTheme(Theme)"/>.
+        /// This will change the appreance of the existing components with property <seealso cref="View.ThemeChangeSensitive"/> on.
+        /// This also affects all components created afterwards.
+        /// </summary>
+        /// <param name="themeId">The theme Id.</param>
+        /// <exception cref="ArgumentNullException">Thrown when the given themeId is null.</exception>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public static void ApplyTheme(string themeId)
+        {
+            if (themeId == null) throw new ArgumentNullException("Invalid themeId");
+
+            int index = customThemes.FindIndex(x => x.Id.Equals(themeId, StringComparison.OrdinalIgnoreCase));
+            if (index >= 0)
+            {
+                CurrentTheme = customThemes[index];
+                return;
+            }
+            
+            index = builtinThemes.FindIndex(x => string.Equals(x.Id, themeId, StringComparison.OrdinalIgnoreCase));
+            if (index >= 0)
+            {
+                CurrentTheme = builtinThemes[index];
+            }
+            else
+            {
+                Tizen.Log.Info("NUI", $"No Theme found with given id : {themeId}");
```

Comment:
What do you think of throwing exception instead of info log?
It will be easier to notify api user that `ApplyTheme` doesn't work with given theme id.

Reply:
Great. I added code to throw an exception when the given themeId is not registered theme.